File: ui/src/types/locationSetting.ts
import { Decoder, array,  boolean,  number, object } from "decoders";

  export interface locationSettingDetails{
    Id:number;
    LocationId :number;
    LastAmcInvoiceNumber  :number;
    LastPaidJobInvoiceNumber  :number;
    LastContractNumber  :number;
    LastReceiptNumber  :number;
    LastSaleInvoiceNumber  :number;
    LastWorkOrderNumber  :number;
  } 

  export const locationSettingDetailsDecoder : Decoder<locationSettingDetails> = object({
    Id:number,
    LocationId :number,
    LastAmcInvoiceNumber  :number,
    LastPaidJobInvoiceNumber  :number,
    LastContractNumber  :number,
    LastReceiptNumber  :number,
    LastSaleInvoiceNumber  :number,
    LastWorkOrderNumber  :number
     });  
  

  export interface locationSettingDetailsResponse{
    LocationSetting:locationSettingDetails,
  }

  export const locationSettingDetailsResponseDecoder: Decoder<locationSettingDetailsResponse> = object({
    LocationSetting: (locationSettingDetailsDecoder),
  });

  export interface LocatioSettingEditResponse {
    isUpdated: Boolean; 
  }
  
  export const locationSettingEditDecoder: Decoder<LocatioSettingEditResponse> = object({
    isUpdated: boolean,
  });
  

File: database/Models/LocationSetting.cs
namespace database.Models
{
    public class LocationSetting
    {
        public int Id { get; set; }
        public int LocationId { get; set; }
        public int LastContractNumber { get; set; }
        public int LastSaleInvoiceNumber { get; set; }
        public int LastPaidJobInvoiceNumber { get; set; }
        public int LastWorkOrderNumber { get; set; }
        public int LastReceiptNumber { get; set; }
    }
}
